core/src/main/java/com/alibaba/fescar/core/protocol/FragmentXID.java

```diff
@@ -0,0 +1,378 @@
+/*
+ *  Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.alibaba.fescar.core.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.util.NetUtil;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import org.apache.commons.lang3.StringUtils;
+import sun.net.util.IPAddressUtil;
+
+/**
+ * FragmentXID is used for support golang distributed cluster server supporting fragmentation
+ *
+ * @author fagongzi(zhangxu19830126 @ gmail.com)
+ */
+public class FragmentXID {
```

Comment:
I think the name XID will better then FragmentXID, since there's no other XID implement